api/bots/poll/poll.py

```diff
@@ -0,0 +1,266 @@
+# See readme.md for instructions on running this code.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from six.moves import range
+
+class PollHandler(object):
+    def usage(self):
+        return '''
+            This plugin maintains up to one poll per user, per topic, in streams.
+            It currently keeps a running count of the votes.
+
+           Design:
+             - polls are visible only within streams;
+             - one poll per user (per topic, per stream);
+             - private messages (or in the stream) control the bot;
+             - the bot only responds in a stream by posting a new or updated poll
+               (and otherwise replies to the sender, eg. for confirmations/errors)
+             - commands:
+                 * new: adds/resets the poll on the stream/topic, with a given title
+                   text, followed by each poll option on a separate line, eg.:
+                     new [stream topic]
+                     How do you feel about this bot?
+                     Terrible!
+                     It's OK.
+                     Great!
+                   This generates a poll with 3 options, defaulting with the current
+                   stream/topic, or using those specified
+                 * vote: adds a vote in a given poll from yourself; you cannot
+                   vote more than once in a given poll. Example command:
+                     vote 99 3 [stream topic]
+                   If a valid index (eg. 1-3 above), the tally for that option is increased,
+                   and the original poll message updated; in this case, poll 99 in the current
+                   stream and topic (unless specified)
+                 * end: ends a poll that you own, allowing you to start a new one, eg.
+                     end [stream topic]
+           BUGS/LIMITATIONS/IDEAS:
+             - *** Currently, stream+topic are space-separated, which rules out spaces in each
+             - When do polls end? What if the poll text can't be edited (a server setting?)
+             - Polls are only in streams - sufficient?
+             - One poll per user per topic - sufficient?
+             - (Once complete, this could migrate to the command-bot API)
+             - Improve help text generally
+             - Support changing votes?
+             - Show votes only at the end?
+             - Other vote 'schemes'?
+             - External vote engines?
+             - External vote storage? (current state is 'maintained' in the message too!)
+            '''
```

Comment:
I think I mentioned somewhere that this was a temporary measure, brainstorming the current state into the same file. I'll move it now; previously I was just looking for some feedback on whether it looked like a good idea before doing more polish.